rfcs/45-stageless.md

```diff
@@ -0,0 +1,977 @@
+        .add_system(gravity.run_if(resource_exists(Gravity)))
+        // The systems in this set won't run if `Paused` is `true`.
+        .add_set(GameSet::Physics.run_if(resource_equals(Paused(false))))
+        .run();
+}
+```
+
+### Exclusive systems
+
+As much as everyone loves to see systems running in parallel, sometimes a little `&mut World` is necessary.
+Previous versions of Bevy treated these "exclusive" systems as a separate concept, with their own set of types and traits.
+They couldn't have `Local` params and could only be inserted at specific points in a frame.
+
+That is no longer the case.
+Exclusive systems are now just regular systems.
+They can have `Local` params and be scheduled wherever you want.
+
+Command application is now signaled through an exclusive system called `apply_system_buffers`.
+You can add instances of this system anywhere in your schedule.
+If one system depends on the effects of commands from another, make sure an `apply_system_buffers` appears somewhere between them.
+
+```rust
+use bevy::prelude::*;
+
+#[derive(SystemLabel)]
+enum MySystems {
+    /* ... */
+    FlushProjectiles,
+    /* ... */
+}
+
+struct ProjectilePlugin;
+
+impl Plugin for ProjectilePlugin {
+    fn build(app: &mut App) {
+        app
+        .add_system(spawn_projectiles.before(CoreSystems::FlushPostUpdate))
+        .add_systems(
+            chain![
+                check_if_projectiles_hit,
+                despawn_projectiles_that_hit,
+                // wherever you want commands to be applied, insert an instance of `apply_system_buffers`
+                apply_system_buffers.named(MySystems::FlushProjectiles)
+                fork_projectiles,
+            ]
+            .after(CoreSystems::FlushPostUpdate)
+            .before(CoreSystems::FlushLast)
+        )
+    }
+}
+```
+
+### Running other systems on-demand with exclusive systems
+
+Exclusive systems have other uses as well.
+
+All systems and system sets added to an `App` are stored within a resource called `Systems`.
+If you want to run a system or system set, you have to extract it from `Systems` first.
+(This lets `Systems` remain accessible when those systems are running).
+
+`Systems` has methods to extract a system or system set given its label.
+When you extract a system set, you actually receive a **schedule**.
+This is a "frozen", executable version of the set, containing all its systems and conditions, along with instructions to run them in the correct order.
+After running a schedule, you can return it and its systems to `Systems`.
+
+The default `App` runner itself actually uses these methods to run the startup sequence and main update loop, so if you retrieve and run a single system somewhere, you'll effectively have written a very basic executor!
+
+```rust
+
+fn example_run_schedule_system(world: &mut World) {
+    // take ownership of the schedule
+    let mut systems = world.resource_mut::<Systems>();

```

Comment:
instead of this
fn example_run_schedule_system(world: &mut World) {
    // take ownership of the schedule
    let mut systems = world.resource_mut::<Systems>();
    let mut schedule = systems.checkout_schedule(MySystemSetLabel);

    // run it on the world
    schedule.run(world);
    
    // return it
    let mut systems = world.resource_mut::<Systems>();
    systems.checkin_schedule(schedule);
}

it'll look something like
fn example_run_schedule_system(world: &mut World) {
    // take ownership of the schedul
    world.schedule_scope(MySystemSetLabel, |world, schedule| {
        
        // run it on the world
        schedule.run(world);
    });
}

if I read the above conversation correctly

From @hymm